src/overlay/RandomPeerSource.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright 2018 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "overlay/RandomPeerSource.h"
+#include "main/Application.h"
+#include "overlay/PeerRecord.h"
+#include "util/Math.h"
+
+#include <algorithm>
+
+namespace stellar
+{
+
+RandomPeerSource::RandomPeerSource(Application& app, size_t batchSize)
+    : mApp(app), mBatchSize(batchSize)
+{
+}
+
+std::vector<PeerRecord>
+RandomPeerSource::getRandomPeers(bool outbound, size_t size,
```

Comment:
this code is wrong:
if I have 1000 peers in the outbound list and this function gets called with size=300 for example, at some point it grabs only 100 peers from the outbound list instead of 300.
This will be a problem because we have 1000 peers to chose from and would still send out junk peers from the inbound table.

it should not wait for peers to get empty before calling loadRandomPeers if it knows it will run out of peers